dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -56,46 +56,52 @@ public ChatHistory(IEnumerable<ChatMessageContent> messages)
     /// <param name="content">Message content</param>
     /// <param name="encoding">Encoding of the message content</param>
     /// <param name="metadata">Dictionary for any additional metadata</param>
+    /// <param name="name">Name of the author of the message</param>
     /// </summary>
-    public void AddMessage(AuthorRole authorRole, string content, Encoding? encoding = null, IReadOnlyDictionary<string, object?>? metadata = null) =>
-        this.Add(new ChatMessageContent(authorRole, content, null, null, encoding, metadata));
+    public void AddMessage(AuthorRole authorRole, string content, Encoding? encoding = null, IReadOnlyDictionary<string, object?>? metadata = null, string? name = null) =>
+        this.Add(new ChatMessageContent(authorRole, content, null, null, encoding, metadata, name));
```

Comment:
This can be problematic if a developer has a dependency on SK and also on a library that uses SK. If the direct and indirect dependencies are for different SK versions and the older version get's loaded first then there is an issue. We've seen this problem in our polyglot notebooks.

What this and the other changes indicates it that all of these helper methods need updating as new properties get added to the `ChatMessageContent`.

We should consider marking the helpers as obsolete and just relying on `ChatHistory.AddMessage(ChatMessageContent chatMessage)`. Developers can create their own extension methods for the additional helpers easily enough and this simplifies our API surface.

@matthewbolanos @stephentoub Thoughts?

Reply:
Obsolete, ok...I like that.

Reply:
Ok...here's the deal.  Adding an overload with a nullable string parameter at the end creates this error for every single caller:

CS0121	The call is ambiguous between the following methods or properties.

I really struggled with adding `name` as the final parameter as it didn't seem like a natural ordering.  We can create an ordering with `name` in a different position.  This will remove the ambiguity, but it also implies we leave the old methods non-[Obsolete] to avoid breaking existing patterns.

If these's are the only methods we're addressing...the scope is contrained enough, extra overloads seem fine.  I'll move to this posture for next review.

**IF** this pattern needs to propagate through a bunch of internal method signatures (which I'm not hearing), I'd be curious to examine those as well.

*If* these methods were returning the message being created (instead of void), the caller would have the option to explicitly add the name or any other state adjustment.  Does changing the return type break the transient dependency case?

Reply:
Pushed update.  

Answered my own question: (as expected) changing return type is a breaking signature change.